contract/contracts/NewAudius.sol

```diff
@@ -0,0 +1,143 @@
+        address tokenSender,
+        string memory campaignInfoCid,
+        string memory recipientsCid,
+        uint32 recipientsNum,
+        uint256 startDate,
+        uint256 endDate,
+        string memory baseURL
+    ) public override {
+        // TODO Update checking tokenSender logic with token issuance phase
+        require(msg.sender == tokenSender, "Token holder must match to msg.sender");
+        uint256 allowance = getAllowanceOf(token, tokenSender);
+        require(allowance > 0, "No token is approved to transfer");
+        require(allowance >= recipientsNum, "Token amount is not enough to distribute");
+
+        uint256 claimAmount = calculateClaimAmount(allowance, recipientsNum);
+        AudiusFollowersCampaign campaign = new AudiusFollowersCampaign(
+            token,
+            campaignInfoCid,
+            recipientsCid,
+            claimAmount,
+            tokenSender,
+            startDate,
+            endDate,
+            baseURL,
+            link
+        );
+        transferToken(token, tokenSender, address(campaign), allowance);
+        campaignList[nextCampaignId] = address(campaign);
+        nextCampaignId = nextCampaignId.add(1);
+        campaign.transferOwnership(msg.sender);
+
+        emit CreateCampaign(
+            token,
+            recipientsCid,
+            recipientsNum,
+            startDate,
+            endDate
+        );
+    }
+}
+
+contract AudiusFollowersCampaign is CampaignInterface {
+    using SafeMath64 for uint64;
+
+    uint64 public nextUserId = 1;
+    mapping(address => uint64) public userIdList;
+    mapping(address => bool) public claimedUserList;
+    mapping(bytes32 => bool) private claimKeyHashList;
+
+    // TODO Integrate with parent constructor
+    constructor(
+        address payable _token,
+        string memory _campaignInfoCid,
+        string memory _recipientsCid,
+        uint256 _claimAmount,
+        address _refundDestination,
+        uint256 _startDate,
+        uint256 _endDate,
+        string memory _baseURL,
+        address _link
+    ) public CampaignInterface(
+        _token,
+        _campaignInfoCid,
+        _recipientsCid,
+        _claimAmount,
+        _refundDestination,
+        _startDate,
+        _endDate,
+        _baseURL,
+        _link
+    ) {}
+
+    function generateClaimKey(uint64 userId) public pure returns (uint256){
+        return uint256(userId) * 10 + 1;
+        // 1 as true
+    }
+
+    function isClaimable() public view override returns (bool) {
+        uint64 userId = userIdList[msg.sender];
+        if (userId == 0) {
+            return false;
+        }
+        if (claimedUserList[msg.sender]) {
+            return false;
+        }
+        uint256 claimKey = generateClaimKey(userId);
+        bytes32 claimKeyHash = keccak256(abi.encodePacked(claimKey));
+        return claimKeyHashList[claimKeyHash];
+    }
+
+    // TODO Logic could be changed
+    function claim() external override {
```

Comment:
To determine if it is Claimable, we need to check the status of the campaign(Status.Cancelled and startDate, endDate).